### PR TITLE
feat: migrate CreateSellOrderForm to react-hook-form

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/rnd-dev-team#1447

- Migrate `CreateSellOrderForm` to react-hook-form

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-2132--regen-marketplace.netlify.app/profile/portfolio

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
While testing the form I realized that newly created sell orders are not retrieved (anything after ID 340) despite the TXs being successful. The problem is also there on https://dev.app.regen.network
This might be a temporary problem with redwood.
/cc @ryanchristo @blushi

**Comment by blushi** [comment]
> While testing the form I realized that newly created sell orders are not retrieved (anything after ID 340) despite the TXs being successful. The problem is also there on https://dev.app.regen.network This might be a temporary problem with redwood. /cc @ryanchristo @blushi

It seems like because we only query for the first page, also see: https://github.com/regen-network/regen-web/blob/dev/web-marketplace/src/pages/Marketplace/Storefront/hooks/useNormalizedSellOrders.tsx#L61-L64
Could that be covered by https://github.com/regen-network/rnd-dev-team/issues/1224 @flagrede?
@ryanchristo could you confirm that sell orders are now ordered by ids in the sell orders query?

File: web-marketplace/src/components/organisms/CreateSellOrderForm/CreateSellOrderForm.constants.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const BASIC_INFO_NAME_LABEL = 'Project name';
+export const BASIC_INFO_NAME_DESCRIPTION =
+  'This is the name of the farm, ranch, property, or conservation project.';
+export const BASIC_INFO_NAME_PLACEHOLDER = 'i.e. Sunnybrook Farms';
+export const BASIC_INFO_SIZE_LABEL = 'Size in hectares or acres';
```
Comment (blushi) [inline web-marketplace/src/components/organisms/CreateSellOrderForm/CreateSellOrderForm.constants.ts:5]: why is this here? probably some leftovers from copying/pasting?

**Review by blushi — APPROVED** [review]
tACK, pre-approving pending clean up

File: web-marketplace/src/components/organisms/CreateSellOrderForm/CreateSellOrderForm.constants.ts (hunk 1 above)
Comment (flagrede) [reply web-marketplace/src/components/organisms/CreateSellOrderForm/CreateSellOrderForm.constants.ts:5]: You're right, will clean up that.

**Comment by flagrede** [comment]
> > While testing the form I realized that newly created sell orders are not retrieved (anything after ID 340) despite the TXs being successful. The problem is also there on https://dev.app.regen.network This might be a temporary problem with redwood. /cc @ryanchristo @blushi
> 
> It seems like this is because we only query for the first page, also see: https://github.com/regen-network/regen-web/blob/dev/web-marketplace/src/pages/Marketplace/Storefront/hooks/useNormalizedSellOrders.tsx#L61-L64 Could that be covered by [regen-network/rnd-dev-team#1224](https://github.com/regen-network/rnd-dev-team/issues/1224) @flagrede? @ryanchristo could you confirm that sell orders are now ordered by ids in the sell orders query?

Oh I forgot about that, yes we could definitely do that in https://github.com/regen-network/rnd-dev-team/issues/1224
